Description: remove wrong qreal check, not needed since Qt 5.2
Author: Dmitry Shachnev <mitya57@gmail.com>
Forwarded: yes, https://codereview.qt-project.org/87066
Last-Update: 2014-06-08

--- a/tests/auto/qgeocoordinate/tst_qgeocoordinate.cpp
+++ b/tests/auto/qgeocoordinate/tst_qgeocoordinate.cpp
@@ -520,32 +520,17 @@ private slots:
             << BRISBANE
             << qreal(1374820.1618767744)
             << qreal(211.1717286649)
-// the following platforms use float for qreal
-#if !defined(QT_NO_FPU) && !defined(Q_PROCESSOR_ARM) && !defined(Q_OS_WINCE) && !defined(Q_OS_SYMBIAN)
             << MELBOURNE;
-#else
-            << QGeoCoordinate(-37.8142515084775, 144.963170622944);
-#endif
         QTest::newRow("london -> new york")
             << LONDON
             << qreal(5570538.4987236429)
             << qreal(288.3388804508)
-// the following platforms use float for qreal
-#if !defined(QT_NO_FPU) && !defined(Q_PROCESSOR_ARM) && !defined(Q_OS_WINCE) && !defined(Q_OS_SYMBIAN)
             << NEW_YORK;
-#else
-            << QGeoCoordinate(40.7145220608416, -74.0071216045375);
-#endif
         QTest::newRow("north pole -> south pole")
             << NORTH_POLE
             << qreal(20015109.4154876769)
             << qreal(180.0)
-// the following platforms use float for qreal
-#if !defined(QT_NO_FPU) && !defined(Q_PROCESSOR_ARM) && !defined(Q_OS_WINCE) && !defined(Q_OS_SYMBIAN)
             << SOUTH_POLE;
-#else
-            << QGeoCoordinate(-89.9999947369857, -90.0);
-#endif
     }
 
     void degreesToString()
--- a/tests/auto/qgeopositioninfo/tst_qgeopositioninfo.cpp
+++ b/tests/auto/qgeopositioninfo/tst_qgeopositioninfo.cpp
@@ -73,21 +73,15 @@ QList<qreal> tst_qgeopositioninfo_qrealTestValues()
 {
     QList<qreal> values;
 
-// the following platforms use float for qreal
-#if !defined(QT_NO_FPU) && !defined(Q_PROCESSOR_ARM) && !defined(Q_OS_WINCE) && !defined(Q_OS_SYMBIAN)
     if (qreal(DBL_MIN) == DBL_MIN)
         values << DBL_MIN;
-#endif
 
     values << FLT_MIN;
     values << -1.0 << 0.0 << 1.0;
     values << FLT_MAX;
 
-// the following platforms use float for qreal
-#if !defined(QT_NO_FPU) && !defined(Q_PROCESSOR_ARM) && !defined(Q_OS_WINCE) && !defined(Q_OS_SYMBIAN)
     if (qreal(DBL_MAX) == DBL_MAX)
         values << DBL_MAX;
-#endif
 
     return values;
 }
--- a/tests/auto/qgeosatelliteinfo/tst_qgeosatelliteinfo.cpp
+++ b/tests/auto/qgeosatelliteinfo/tst_qgeosatelliteinfo.cpp
@@ -73,21 +73,15 @@ QList<qreal> tst_qgeosatelliteinfo_qrealTestValues()
 {
     QList<qreal> values;
 
-// the following platforms use float for qreal
-#if !defined(QT_NO_FPU) && !defined(Q_PROCESSOR_ARM) && !defined(Q_OS_WINCE) && !defined(Q_OS_SYMBIAN)
     if (qreal(DBL_MIN) == DBL_MIN)
         values << DBL_MIN;
-#endif
 
     values << FLT_MIN;
     values << -1.0 << 0.0 << 1.0;
     values << FLT_MAX;
 
-// the following platforms use float for qreal
-#if !defined(QT_NO_FPU) && !defined(Q_PROCESSOR_ARM) && !defined(Q_OS_WINCE) && !defined(Q_OS_SYMBIAN)
     if (qreal(DBL_MAX) == DBL_MAX)
         values << DBL_MAX;
-#endif
 
     return values;
 }
